src/extensions/idle.rs

```diff
@@ -196,73 +191,55 @@ impl<'a, T: Read + Write + 'a> Handle<'a, T> {
             (_, result) => result,
         }
     }
-
-    /// Block until the given callback returns `false`, or until a response
-    /// arrives that is not explicitly handled by [`UnsolicitedResponse`].
-    pub fn wait_while<F>(mut self, callback: F) -> Result<()>
-    where
-        F: FnMut(UnsolicitedResponse) -> bool,
-    {
-        self.wait_inner(true, callback).map(|_| ())
-    }
 }
 
 impl<'a, T: SetReadTimeout + Read + Write + 'a> Handle<'a, T> {
-    /// Set the keep-alive interval to use when `wait_keepalive_while` is called.
+    /// Set the timeout duration on the connection. This will also set the frequency
+    /// at which the connection is refreshed.
+    ///
+    /// The interval defaults to 29 minutes as given in RFC 2177.
+    pub fn set_timeout(&mut self, interval: Duration) -> &mut Self {
```

Comment:
I think [the convention for builders](https://rust-lang.github.io/api-guidelines/type-safety.html#c-builder) is to not include the `set_` prefix.

Reply:
Oh, I didn't know this existed! Thanks!